CONTRIBUTING.md

```diff
@@ -4,46 +4,97 @@ You are most welcome to create pull requests and issues to ecChronos.
 Before creating pull requests it is recommended to first discuss your idea with at least one of the owners of the repository.
 For keeping track of the history it is recommended that most communication is performed or summarized within pull requests or issues.
 
-## Branches
+## Development
 
-Target the minimum possible branch version when fixing an issue.
-New features should in general be added to the master branch.
+### Prerequisites
 
-## Pull requests
+* JDK8
+* Docker (for test setup)
 
-1. Ensure that the pull request is targeted at the minimum possible version based on the type of change.
-2. If you intend to fix an issue,
- mention the issue number in the text body of the pull requests, i.e. `Fixes #20`.
-3. Add a changelog entry at the bottom of the current target version when handling an issue.
-4. All checks must pass before merging a pull request.
-5. In general at least one project admin should approve the pull request before merging.
+### Branches
 
-## Merging
+Target the lowest applicable version of ecChronos when fixing an issue.
+Bug fixes should be targeted to the lowest maintained version where the bug reside.
+New features should, in general, be added to the master branch.
 
-Merging is performed from minimum version towards master.
-It is the responsibility of the person merging the pull request to make sure it gets merged to master.
-
-## Code Style
+### Code Style
 
 Codestyle TBD.
+
 It is appreciated if provided patches are contained and do not modify code outside of the scope of the patch.
 This will also make it quicker to perform reviews and merging the pull requests.
 
-## Builds & Tests
-
+### Builds
+    
 If you're looking to contribute to ecChronos it is recommended that you sign up on [Travis](https://travis-ci.org/) and [Coveralls](https://coveralls.io/) and enable builds and reports on your own fork of ecChronos.
 The builds required to merge a pull requests are contained within the [Travis configuration](.travis.yml) and include tests, code coverage as well as PMD checks.
 
+All checks needs to pass before merging a pull request.
+Sometimes the PMD rules does not match the goal of the project and in those cases it might be relevant to discuss removing or ignoring specific rules/violations.
+Currently there is a base of PMD rules used and that will evolve with the project,
+so if a rule seems to be detrimental rather helping the project,
```

Comment:
...rather than helping...?

Reply:
I think it's technically correct (the best kind) as the PMD rules are there to help the project. But I see your point so I rephrased this whole segment. :)

Reply:
I was mostly referring to the missing "than", but this work too. :-)